index.js

```diff
@@ -10,12 +10,45 @@
  * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
  * and limitations under the License.
  */
-var chalk = require('chalk');
-var GroupMessages = require('./lib/utils/groupMessages');
-var TEMPLATE_DEPRECATION_WARNINGS = GroupMessages.GROUP.TemplateDeprecationWarnings;
-var REGISTER_TEMPLATE_DEPRECATION_WARNINGS =
-  GroupMessages.GROUP.RegisterTemplateDeprecationWarnings;
-var SASS_MAP_FORMAT_DEPRECATION_WARNINGS = GroupMessages.GROUP.SassMapFormatDeprecationWarnings;
+
+// TODO: Add changelog entry about removed deprecated features
+// import chalk from 'chalk';
+// import GroupMessages from './lib/utils/groupMessages';
+import extend from './lib/extend.js';
+import transform from './lib/common/transforms.js';
+import transformGroup from './lib/common/transformGroups.js';
+import format from './lib/common/formats.js';
+import action from './lib/common/actions.js';
+import * as formatHelpers from './lib/common/formatHelpers/index.js';
+import filter from './lib/common/filters.js';
+import registerTransform from './lib/register/transform.js';
+import registerTransformGroup from './lib/register/transformGroup.js';
+import registerFormat from './lib/register/format.js';
+import registerTemplate from './lib/register/template.js';
+import registerAction from './lib/register/action.js';
+import registerFilter from './lib/register/filter.js';
+import registerParser from './lib/register/parser.js';
+import registerFileHeader from './lib/register/fileHeader.js';
+import exportPlatform from './lib/exportPlatform.js';
+import buildPlatform from './lib/buildPlatform.js';
+import buildAllPlatforms from './lib/buildAllPlatforms.js';
+import cleanPlatform from './lib/cleanPlatform.js';
+import cleanAllPlatforms from './lib/cleanAllPlatforms.js';
+
+// const TEMPLATE_DEPRECATION_WARNINGS = GroupMessages.GROUP.TemplateDeprecationWarnings;
+// const REGISTER_TEMPLATE_DEPRECATION_WARNINGS = GroupMessages.GROUP.RegisterTemplateDeprecationWarnings;
+// const SASS_MAP_FORMAT_DEPRECATION_WARNINGS = GroupMessages.GROUP.SassMapFormatDeprecationWarnings;
+
+// Allow to be overridden by setter, set default to memfs for browser env, node:fs for node env
+export let fs =
+  typeof window === 'object'
+    ? (await import('@bundled-es-modules/memfs')).default
+    : (await import('node:fs')).default;
+
+// since ES modules exports are read-only, use a setter
+export const setFs = (_fs) => {
+  fs = _fs;
+};
```

Comment:
important change, to allow customizing the filesystem that's used.

- If browser -> use memfs by default
- If node -> use node:fs by default
- Allow setting it to a different FS shim/equivalent with setFs()